snuba/reader.py

```diff
@@ -110,3 +119,43 @@ def execute(
             ),
             with_totals=with_totals,
         )
+
+
+class HTTPReader(Reader):
+    def __init__(
+        self, host: str, port: int, settings: Optional[Mapping[str, str]] = None
+    ):
+        if settings is not None:
+            assert "query_id" not in settings, "query_id cannot be passed as a setting"
+        self.__pool = HTTPConnectionPool(host, port)
+        self.__default_settings = settings if settings is not None else {}
+
+    def execute(
+        self,
+        query: str,
+        settings: Optional[Mapping[str, str]] = None,
+        query_id: Optional[str] = None,
+        with_totals: bool = False,  # NOTE: unnecessary with FORMAT JSON
+    ) -> Result:
+        if settings is None:
+            settings = {}
+
+        assert "query_id" not in settings, "query_id cannot be passed as a setting"
+        if query_id is not None:
+            settings["query_id"] = query_id
+
+        response = self.__pool.urlopen(
+            "POST",
+            "/?" + urlencode({**self.__default_settings, **settings}),
+            body=query.encode("utf-8"),
+        )
+
+        # TODO: Distinguish between ClickHouse errors and other HTTP errors.
+        if response.status // 100 != 2:
+            raise HTTPError(f"{response.status} Unexpected")
```

Comment:
I think we should include the error message in the response payload before this goes to production otherwise we would not be able to see Clickhouse errors in prod.
Also not having the clickhouse error would make things very hard to troubleshoot if something goes wrong when you roll this out.